src/main/java/duke/commands/DukeCommand.java

```diff
@@ -0,0 +1,70 @@
+package duke.commands;
+
+import duke.exceptions.DukeException;
+import duke.exceptions.DukeExceptionCommandNotFound;
+import duke.exceptions.DukeExceptionIllegalArgument;
+import duke.storage.FileLoader;
+import duke.tasks.Event;
+import duke.tasks.Todo;
+import duke.tasks.Deadline;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+import duke.ui.Ui;
+
+public abstract class DukeCommand {
+
+    public boolean isExit() {
+        return false;
+    }
+
+    /**
+     * Returns command based on user input line.
+     *
+     * The first-hop parsing occurs here.
+     * The rest of the parsing is offloaded to the individual parsing classes.
+     *
+     * @param line User input string.
+     * @return DukeCommand.
+     * @throws DukeExceptionCommandNotFound When command supplied is invalid.
+     * @throws DukeExceptionIllegalArgument When task parsing error occurs.
+     */
+    public static DukeCommand parse(String line)
+            throws DukeExceptionCommandNotFound, DukeExceptionIllegalArgument {
+        line = line.strip(); // input sanitization
+        String arg = "", cmd = line;
+        if (line.contains(" ")) {
+            int splitIdx = line.indexOf(' ');
+            cmd = line.substring(0, splitIdx); // automatically stripped
+            arg = line.substring(splitIdx+1).strip();
```

Comment:
Should the + be separated by spaces? A few other instances of this around too.

Reply:
Yes